packages/logger/src/Logger.ts

```diff
@@ -493,27 +511,31 @@ class Logger extends Utility implements LoggerInterface {
   }
 
   /**
-   * Alias for removePersistentLogAttributes.
+   * It removes attributes based on provided keys to all log items generated by this Logger instance.
    *
    * @param {string[]} keys
    * @returns {void}
    */
   public removeKeys(keys: string[]): void {
-    this.removePersistentLogAttributes(keys);
+    const removeKey = (attributes: LogAttributes, key: string): void => {
+      if (attributes && key in attributes) {
+        delete attributes[key];
+      }
+    };
+    for (const key of keys) {
+      removeKey(this.persistentLogAttributes, key);
+      removeKey(this.temporaryLogAttributes, key);
+    }
   }
 
   /**
-   * It removes attributes based on provided keys to all log items generated by this Logger instance.
+   * Alias for removeKeys().
    *
    * @param {string[]} keys
    * @returns {void}
    */
   public removePersistentLogAttributes(keys: string[]): void {
-    for (const key of keys) {
```

Comment:
I think this should not be an alias but instead remove only from `persistentLogAttributes`.